shuup/admin/views/wizard.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2016, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import six
+from django.db.transaction import atomic
+from django.http.response import JsonResponse
+from django.utils.datastructures import OrderedDict
+from django.utils.translation import ugettext_lazy as _
+
+from shuup import configuration
+from shuup.admin.form_part import FormPartsViewMixin, TemplatedFormDef
+from shuup.admin.utils.views import CreateOrUpdateView
+from shuup.core.models import Shop
+from shuup.utils.form_group import FormDef, FormGroup
+from shuup.utils.iterables import first
+
+GENERAL_INFO_PANE_TYPE = 1
+PAYMENT_METHOD_PANE_TYPE = 2
+SHIPPING_METHOD_PANE_TYPE = 3
+THEME_PANE_TYPE = 4
+FINAL_PANE_TYPE = 5
+
+WIZARD_PANES = [
+    (GENERAL_INFO_PANE_TYPE, _("General Information"), "fa fa-info-circle"),
+    (PAYMENT_METHOD_PANE_TYPE, _("Payment Methods"), "fa fa-money"),
+    (SHIPPING_METHOD_PANE_TYPE, _("Shipping Methods"), "fa fa-truck"),
+    (THEME_PANE_TYPE, _("Themes"), "fa fa-picture-o"),
+    (FINAL_PANE_TYPE, "", "")
+]
+
+
+class WizardFormDefMixin(object):
+    def __init__(self, pane_identifier,  **kwargs):
+        self.pane_identifier = pane_identifier
+        self.context = kwargs.pop("context", {})
+        self.extra_js = kwargs.pop("extra_js", "")
+        super(WizardFormDefMixin, self).__init__(**kwargs)
+
+
+class WizardFormDef(WizardFormDefMixin, FormDef):
+    pass
+
+
+class TemplatedWizardFormDef(WizardFormDefMixin, TemplatedFormDef):
+    pass
+
+
+class _WizardPane(FormGroup):
+    def __init__(self, identifier, title, icon, **kwargs):
+        super(_WizardPane, self).__init__(**kwargs)
+        self.identifier = identifier
+        self.title = title
+        self.icon = icon
+
+    @atomic
+    def save(self):
+        for form in six.itervalues(self.forms):
+            form.save()
+
+
+class WizardView(FormPartsViewMixin, CreateOrUpdateView):
+    model = Shop
+    context_object_name = "shop"
+    template_name = "shuup/admin/wizard/wizard.jinja"
+    form_part_class_provide_key = "admin_setup_wizard_form_part"
+    wizard_panes = [
```

Comment:
There is something weid about this. I think these should be defined in settings. And for example the shop module could introduce the `GENERAL_INFO_PANE_TYPE` and `FINAL_PANE_TYPE`.